### PR TITLE
Various doc updates

### PR DESCRIPTION
**Pull request by wvandeun**
- fixes the groups guide
- use hfid in repository guide to update repository ref
- use hfid in checks guide
- improve the graphql topic (add information about hfid, upsert mutations and delete mutation)

### CODE REVIEW

File: docs/docs/guides/repository.mdx [hunk 1]
```diff
@@ -247,7 +247,7 @@ Unlike `Repository`, Infrahub does not automatically update `Read-only Repositor
   mutation {
     CoreReadOnlyRepositoryUpdate(
       data: {
-        id: "ID_OF_THE_REPOSITORY"
+        hfid: ["My Git repository"]
```
Comment (BaptisteGi) [inline docs/docs/guides/repository.mdx:250]: @wvandeun are you sure we need to put the HFID inside brackets? earlier in this doc there is an example with credentials > repository using HFID that looks like this: `credential: {hfid: "my-git-credential"}`. I think I tested back then and it was working fine

File: docs/docs/topics/graphql.mdx [hunk 2]
```diff
@@ -15,31 +15,33 @@ If you need to extract the current GraphQL schema in your environment you can is
...
-                __typename
-            }
+  CoreRepository {            # PaginatedCoreRepository object
+    count
+      edges {                 # EdgedCoreRepository object
+        node {                # CoreRepository object
+          id
+          hfid
+          display_label
+          __typename
```
Comment (BaptisteGi) [inline docs/docs/topics/graphql.mdx]:
```suggestion
  CoreRepository {            # PaginatedCoreRepository object
    count
    edges {                 # EdgedCoreRepository object
      node {                # CoreRepository object
        id
        hfid
        display_label
        __typename
```

**Review by BaptisteGi — APPROVED** [review]
looks good 👍

File: docs/docs/topics/graphql.mdx (hunk 2 above)
Comment (wvandeun) [reply docs/docs/topics/graphql.mdx:41]: You do not need to use the brackets, as long as the hfid is made up of a single field. If the hfid is made up of multiple fields, you HAVE to use brackets.

My personal opinion is that we should keep things simple and only have one way of doing things and therefor I prefer to always use the brackets.

Not sure if there's other opinions on that?

File: docs/docs/guides/repository.mdx (hunk 1 above)
Comment (wvandeun) [reply docs/docs/guides/repository.mdx:250]: You don't have to use brackets as long as the HFID is made up of a single field. You must use brackets when the HFID is made up of multiple fields. 

My personal opinion is that we should keep things simple and that we should only have one way of doing something. So I prefer to always use brackets. 

Not sure if there's any other opinions on this. Maybe should even remove the variant where you don't have to provide the brackets. IMO it is only providing confusion for users.

**Pull request merged**